spring-cloud-stream/src/main/java/org/springframework/cloud/stream/config/ChannelBindingServiceProperties.java

```diff
@@ -100,128 +136,36 @@ public void setDynamicDestinations(String[] dynamicDestinations) {
-					String.valueOf(bindingProperties.getBatchSize()));
-		}
-		if (bindingProperties.getBatchBufferLimit() != null) {
-			producerProperties.setProperty(BinderPropertyKeys.BATCH_BUFFER_LIMIT,
-					String.valueOf(bindingProperties.getBatchBufferLimit()));
-		}
-		if (bindingProperties.getBatchTimeout() != null) {
-			producerProperties.setProperty(BinderPropertyKeys.BATCH_TIMEOUT,
-					String.valueOf(bindingProperties.getBatchTimeout()));
-		}
+	public void setProducerDefaults(Properties producerDefaults) {
+		this.producerDefaults = producerDefaults;
 	}
 
-	private void updateProducerPartitionProperties(BindingProperties bindingProperties, Properties producerProperties) {
-		if (isPartitionedProducer(bindingProperties)) {
-			if (bindingProperties.getPartitionKeyExpression() != null) {
-				producerProperties.setProperty(BinderPropertyKeys.PARTITION_KEY_EXPRESSION,
-						bindingProperties.getPartitionKeyExpression());
-			}
-			if (bindingProperties.getPartitionKeyExtractorClass() != null) {
-				producerProperties.setProperty(BinderPropertyKeys.PARTITION_KEY_EXTRACTOR_CLASS,
-						bindingProperties.getPartitionKeyExtractorClass());
-			}
-			if (bindingProperties.getPartitionSelectorClass() != null) {
-				producerProperties.setProperty(BinderPropertyKeys.PARTITION_SELECTOR_CLASS,
-						bindingProperties.getPartitionSelectorClass());
-			}
-			if (bindingProperties.getPartitionSelectorExpression() != null) {
-				producerProperties.setProperty(BinderPropertyKeys.PARTITION_SELECTOR_EXPRESSION,
-						bindingProperties.getPartitionSelectorExpression());
-			}
-			if (bindingProperties.getPartitionCount() != null) {
-				producerProperties.setProperty(BinderPropertyKeys.NEXT_MODULE_COUNT,
-						Integer.toString(bindingProperties.getPartitionCount()));
-			}
-			if (bindingProperties.getNextModuleConcurrency() != null) {
-				producerProperties.setProperty(BinderPropertyKeys.NEXT_MODULE_CONCURRENCY,
-						Integer.toString(bindingProperties.getNextModuleConcurrency()));
-			}
-		}
+	@Override
+	public void setApplicationContext(ApplicationContext applicationContext) throws BeansException {
+		this.applicationContext = (ConfigurableApplicationContext) applicationContext;
 	}
 
-	private void updateConsumerPartitionProperties(String inputChannelName, Properties consumerProperties) {
-		BindingProperties bindingProperties = this.bindings.get(inputChannelName);
-		if (bindingProperties != null) {
-			if (isPartitionedConsumer(inputChannelName)) {
-				consumerProperties.setProperty(BinderPropertyKeys.COUNT,
-						Integer.toString(getInstanceCount()));
-				consumerProperties.setProperty(BinderPropertyKeys.PARTITION_INDEX,
-						Integer.toString(getInstanceIndex()));
-			}
-		}
+	@Override
+	public void afterPropertiesSet() throws Exception {
+		DefaultConversionService conversionService = new DefaultConversionService();
+		conversionService.addConverter(new SpelExpressionConverterConfiguration.SpelConverter());
```

Comment:
should this be injected as a bean, and/or should we consider annotating that converter bean in the config itself with `@IntegrationConverter`... could then rely on the `integrationConversionService` and the converter would already be registered